src/main/java/hudson/plugins/active_directory/ActiveDirectoryUnixAuthenticationProvider.java

```diff
@@ -127,6 +156,27 @@ public ActiveDirectoryUnixAuthenticationProvider(ActiveDirectorySecurityRealm re
 
         this.userCache = cache.getUserCache();
         this.groupCache = cache.getGroupCache();
+
+        String ldapConnectTimeout;
+        String ldapReadTimeout;
+        if (realm.environmentProperties != null) {
+            Map<String, String> extraEnvVarsMap = ActiveDirectorySecurityRealm.EnvironmentProperty.toMap(realm.environmentProperties);
+            if (!extraEnvVarsMap.containsKey(LDAP_CONNECT_TIMEOUT)) {
```

Comment:
you can actually do all of this with 3 lines of code  (just call `putAll` with the map from the realm - and make the map return an empty map for null input).

```java
 props.put(LDAP_CONNECT_TIMEOUT, System.getProperty(LDAP_CONNECT_TIMEOUT, DEFAULT_LDAP_CONNECTION_TIMEOUT));
props.put(LDAP_READ_TIMEOUT, System.getProperty(LDAP_READ_TIMEOUT, DEFAULT_LDAP_READ_TIMEOUT);
props.putAll(ActiveDirectorySecurityRealm.EnvironmentProperty.toMap(realm.environmentProperties));